server/src/main/java/org/elasticsearch/repositories/RepositoriesService.java

```diff
@@ -411,14 +415,21 @@ private Repository createRepository(RepositoryMetaData repositoryMetaData, Map<S
             throw new RepositoryException(repositoryMetaData.name(),
                 "repository type [" + repositoryMetaData.type() + "] does not exist");
         }
+        boolean success = false;
```

Comment:
This isn't strickly necessary with the changes just now, but maybe fine to keep it here since we discovered it here and I think we should def. cleanup on a failed `start` out of principle?